modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java

```diff
@@ -1838,6 +1838,327 @@ public void testPublishCustomObjectDefinition() throws Exception {
 		}
 	}
 
+	@Test
+	public void testPublishNode() throws Exception {
+		TreeTestUtil.createObjectDefinitionTree(
+			LinkedHashMapBuilder.put(
+				"A", new String[] {"AA"}
+			).put(
+				"AA", new String[0]
+			).build(),
+			_objectDefinitionLocalService, Arrays.asList("A"),
+			_objectRelationshipLocalService);
```

Comment:
Hey @MarinhoFeliphe,
We discussed changing this method `TreeTestUtil.createObjectDefinitionTree()` to create a Tree (or draft or publish). Besides that, analyze if it is necessary to:
- create a test method that will assert the created Tree. This method would be private in this class to avoid repeating the same code block (creating a tree and asserting if it is correct).

Reply:
@marcelabc done [here](https://github.com/liferay-objects/liferay-portal/pull/3694/commits/f0ab4bbde838fa763769e2fdf425734a48b288b4)